code/modules/cooking/machinery/cooking_machines/_cooker.dm

```diff
@@ -0,0 +1,205 @@
+			else
+				description += "It is running at [round(get_efficiency(), 0.1)]% efficiency!"
+			description += "<br>Temperature: [round(temperature - T0C, 0.1)]C / [round(optimal_temp - T0C, 0.1)]C"
+		else
+			description += "It is switched off."
+		openToolTip(usr, src, params, name, description)
+
+/obj/machinery/appliance/cooker/MouseExited(location, control, params)
+	. = ..()
+	closeToolTip(usr)
+
+/obj/machinery/appliance/cooker/list_contents(var/mob/user)
+	if (length(cooking_objs))
+		var/string = "Contains...</br>"
+		var/num = 0
+		for (var/a in cooking_objs)
+			var/datum/cooking_item/CI = a
+			num++
+			if (CI && CI.container)
+				string += "- [CI.container.label(num)], [report_progress(CI)]</br>"
+		to_chat(usr, string)
+	else
+		to_chat(usr, SPAN_NOTICE("It's empty."))
+
+/obj/machinery/appliance/cooker/proc/get_efficiency()
+	. = (cooking_power / optimal_power) * 100
+
+/obj/machinery/appliance/cooker/Initialize()
+	. = ..()
+	var/interval = (optimal_temp - min_temp)/temp_settings
+	for(var/newtemp = min_temp - interval, newtemp<=optimal_temp, newtemp+=interval)
+		var/image/disp_image = image('icons/screen/radial.dmi', "radial_temp")
+		var/hue = RotateHue(hsv(0, 255, 255), 120 * (1 - (newtemp-min_temp)/(optimal_temp-min_temp)))
+		disp_image.color = HSVtoRGB(hue)
+		temp_options["[newtemp - T0C]"] = disp_image
+	temp_options["OFF"] = image('icons/misc/mark.dmi', "x3")
+	loss = (active_power_usage / resistance)*0.5
+	cooking_objs = list()
+	for(var/cctype in starts_with)
+		if (length(cooking_objs) >= max_contents)
+			break
+		if(isnum(starts_with[cctype]))
+			for(var/i = 0, i<starts_with[cctype], i++)
+				cooking_objs.Add(new /datum/cooking_item/(new cctype(src)))
+		else
+			cooking_objs.Add(new /datum/cooking_item/(new cctype(src)))
+	cooking = FALSE
+
+	queue_icon_update()
+
+/obj/machinery/appliance/cooker/attempt_toggle_power(mob/user)
+	var/old_power = use_power
+	if (!isliving(user))
+		return
+	if (!user.check_dexterity(DEXTERITY_SIMPLE_MACHINES))
+		return
+	if (user.stat || user.restrained() || user.incapacitated())
+		return
+	if (!Adjacent(user) && !issilicon(user))
+		to_chat(user, "You can't reach [src] from here.")
+		return
+
+	var/desired_temp = show_radial_menu(user, src, temp_options - (!old_power ? "OFF" : "[set_temp-T0C]"), require_near = TRUE, tooltips = TRUE, no_repeat_close = TRUE)
+	if(!desired_temp)
+		return
+
+	if(desired_temp == "OFF")
+		update_use_power(POWER_USE_OFF)
+	else
+		set_temp = text2num(desired_temp) + T0C
+		to_chat(user, SPAN_NOTICE("You set [src] to [round(set_temp-T0C)]C."))
+		update_use_power(POWER_USE_IDLE) // will be updated later
+	if(old_power != use_power)
+		activation_message(user)
+	playsound(src, 'sound/machines/click.ogg', 40, 1)
+	update_icon()
+
+/obj/machinery/appliance/cooker/proc/activation_message(var/mob/user)
+	user.visible_message("<b>[user]</b> turns [use_power ? "on" : "off"] [src].", "You turn [use_power ? "on" : "off"] [src].")
```

Comment:
`\The [user]`

Reply:
Not necessary (it's auto-inserted), but I'll include it to make it explicit.